packages/flutter/test/material/text_field_test.dart

```diff
@@ -380,6 +381,58 @@ void main() {
     expect(textField.cursorRadius, const Radius.circular(3.0));
   });
 
+  testWidgets('text field selection toolbar renders correctly inside opacity', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: Container(
+              width: 100,
+              height: 100,
+              child: const Opacity(
+                opacity: 0.5,
+                child: TextField(
+                  decoration: InputDecoration(hintText: 'Placeholder'),
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await tester.showKeyboard(find.byType(TextField));
+
+    const String testValue = 'A B C';
+    tester.testTextInput.updateEditingValue(
+        const TextEditingValue(
+          text: testValue
+        )
+    );
+    await tester.pump();
+
+    // The selectWordsInRange with SelectionChangedCause.tap seems to be needed to show the toolbar.
+    // (This is true even if we provide selection parameter to the TextEditingValue above.)
+    final EditableTextState state = tester.state<EditableTextState>(find.byType(EditableText));
+    state.renderEditable.selectWordsInRange(from: const Offset(0, 0), cause: SelectionChangedCause.tap);
+    await tester.pump();
+
+    expect(state.showToolbar(), true);
+    await tester.pump();
+
+    // This is needed for the AnimatedOpacity to turn from 0 to 1 so the toolbar is visible.
+    await tester.pumpAndSettle();
```

Comment:
it's more efficient to just pump once with the required amount of time to skip the animation

Reply:
I've removed all the previous pumps. However, I couldn't change this `pumpAndSettle` to even `pump(Duration(seconds: 100))` to pass the test. It'll pass with `await tester.pump(); await tester.pump(Duration(seconds: 1));` though. Is this a bug or an expected behavior?

Reply:
I've also added the first `pump` back in the newest patch. Otherwise, the "COPY/CUT/PASTE" will appear in a different order which causes a golden image mismatch (see attached images). Is this also an expected behavior?
![text_field_opacity_test 0 0](https://user-images.githubusercontent.com/22987568/56320950-30f64280-611a-11e9-9159-7186ea23d862.png)
![text_field_opacity_test 0 0 original](https://user-images.githubusercontent.com/22987568/56320951-32c00600-611a-11e9-9b98-e03aee3ffb87.png)

Reply:
No idea why the order would change...

Reply:
looks like you still have pumpAndSettle?

Reply:
Yes, I'll either need this `pumpAndSettle`, or `await tester.pump(); await tester.pump(Duration(seconds: 1));` as mentioned in https://github.com/flutter/flutter/pull/31097/files#r276410094. Do you want me to change to `await tester.pump(); await tester.pump(Duration(seconds: 1));`?

Reply:
it's significantly preferred to use pump than pumpAndSettle, because pumpAndSettle will mask regressions.

Reply:
Changed to `pump`.

Reply:
One suggestion is to capture the return value of `pumpAndSettle` and `expect` the result of it (or expect that it's less than some sane value).

Reply:
I guess the regression that @Hixie mentioned is about the regression in seconds (e.g., the animation should finish in 1 second but a regression made it 10 seconds). I'm not sure if the return value of `pumpAndSettle` would catch that. What's your thought, @Hixie ?

Reply:
The return value would catch it but it's more likely that someone fixing it will just change the number and it's more likely that a reviewer will fail to notice that.